src/leaflike/user/auth.clj

```diff
@@ -0,0 +1,57 @@
+(ns leaflike.user.auth
+  (:require [buddy.hashers :as hashers]
+            [ring.util.response :as res]))
+
+(defonce user-session (atom nil))
+
+(defn- throw-unauthorized
+  [status]
+  (do (when-not (nil? @user-session)
+        (reset! user-session nil))
+      (res/redirect "/login")))
+
+(defn wrap-authorized
+  [handler]
+  (fn [request]
+    (if (nil? @user-session)
+      (throw-unauthorized 401)
+      (handler request))))
+
+(defn wrap-unauthorized
+  [handler]
+  (fn [request]
+    (let [response (handler request)]
+      (when (nil? @user-session)
+        (throw-unauthorized 403))
+      response)))
+
+(defn logout-auth
+  [request]
+  (let [session (:session request)]
+    (if session
+      (do (res/redirect "/login")
+          (reset! user-session nil))
+      (throw-unauthorized 401))))
+
+(defn login-auth
+  [request member]
```

Comment:
Is the request arg getting used> If not, it can be removed. Or if the call needs to pass it, it can be represented as "_" here